File: Proyecto-PracticasProfesionales/backend/interfaces/api/schemas/notificacion.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class NotificacionCreate(BaseModel):
    id_usuario: int
    titulo: str
    mensaje: str


class NotificacionUpdate(BaseModel):
    leida: bool


class NotificacionResponse(BaseModel):
    id_notificacion: int
    id_usuario: int
    titulo: str
    mensaje: str
    leida: bool
    fecha_envio: datetime

    model_config = ConfigDict(from_attributes=True)

File: Proyecto-PracticasProfesionales-main-corregido/src/domain/notificaciones/Notificacion.ts
export interface Notificacion {
  id_notificacion: number;
  id_usuario: number;
  titulo: string;
  mensaje: string;
  leida: boolean;
  fecha_envio: string;
  fecha_creacion?: string;
  tipo?: string;
  categoria?: string;
  prioridad?: string;
  modulo?: string | null;
  entidad?: string | null;
  id_entidad?: number | null;
  url_destino?: string | null;
}


export interface ResumenNotificaciones {
  total: number;
  no_leidas: number;
  leidas: number;
}
